lumispy/utils/analysis.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2022 The LumiSpy developers
+#
+# This file is part of LumiSpy.
+#
+# LumiSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the license, or
+# (at your option) any later version.
+#
+# LumiSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with LumiSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import hyperspy.api as hs
+
+
+def plot_span_map(sig, nspans=1):
+    """
+    Plot a span map.
+    
+    Plots a navigator consisting of the sum over all positions in sig, with `nspans` spans. 
+
+    For each span a `BaseSignal` image is plotted consisting of integrated counts over the the range defined by the span.
+
+    The spans can be moved interactively and the corresponding images will update automatically.
+
+    Arguments
+    ---------
+    sig: hyperspy.signals.Signal1D
+        hyperspectra to inspect. Should have 2 spatial dimensions and 1 signal dimension.
+    nspans: int
+        number of channels to plot. Maximum is 3, default is 1.
+
+    Returns
+    -------
+    all_sum: hs.signals.BaseSignal
+        Sum over all positions of `sig`, the 'navigator' for `plot_span_map`
+    spans: hs.roi.SpanROI
+        The span ROIs from the navigator
+    span_sigs: hs.signals.BaseSignal
+        slices of `sig` according to each span roi
+    span_sums: hs.signals.BaseSignal
+        the summed `span_sigs`.
+    """
+    if nspans > 3:
+        raise ValueError("Maximum number of spans allowed is 3")
```

Comment:
Is there a reason why 3 is the limit?

Reply:
If so, it should be stated in the documentation.

Reply:
Err not especially. It's just that beyond 3, it's a bit more challenging to come up with a sensible colour scheme. 

Using `'red'`, `'green'` and `'blue'` is helpful because we can map these directly to corresponding cmaps: `Reds`, `Greens` and `Blues`: 

https://matplotlib.org/stable/tutorials/colors/colormaps.html?highlight=reds#sequential

Looks like in theory could add purple, orange and gray.

To allow for arbitrary colour combinations would be possible, but would allow for some smart code to map between span colours and appropriate cmaps.

Reply:
Could you try to implement that, following the hyperspy plot colourmaps [here](http://hyperspy.org/hyperspy-doc/current/user_guide/visualisation.html#id28). I believe you can set as default n=3 with RGB colours, or if n>3, then apply the `'mpl_colors' camp. Or if the user sets their own, you can convert the continuous cmap to a discrete n colour list (easily done with matplotlib palets functions.

Reply:
I guess here nothing really prevents invoking twice `plot_span_map` on the same data, no? I guess the interactivity is lost in that case